### PR TITLE
Referrals: send pass share

### PR DESCRIPTION
**Pull request by SergioEstevao**
| 📘 Part of: #2083  |  
|:---:|

Fixes # 

This PR adds the share capability for the Referrals send pass screen.

| 1 | 2 | 3 | 4 | 5 |
| - | - | - | - | - |
| IMG_0145 |IMG_0144 | IMG_0143 | IMG_0141 | IMG_0142 |

I initially tried to wrap the UIActivityViewController on an SwiftUI View but was having issues with the presentation using a bottom sheet ( there was always a part of the bottom sheet that didn't get the right color)

I tried to user ShareLink but the functionality is iOS 16 and above only so I needed to replicate it using UIActivityViewController any way.

In the end I'm using the UIActivityViewController in the host VC for the SendPass View. This allow me to proper display it and keep it contained inside the SendPass flow.

## To test

1. Start the app
2. Ensure you the Referrals FF enabled and you are using a Plus or Patron account
3. Go to Profile
4. Tap on the gift icon on the top left
5. Tap On Share Guest Pass
6. See the Share system view
7. Choose different types of Share and check if they work:
8. Share and see if you go back to the profile screen.

## Checklist

- [x] I have considered if this change warrants user-facing release notes and have added them to `CHANGELOG.md` if necessary.
- [x] I have considered adding unit tests for my changes.
- [x] I have updated (or requested that someone edit) [the spreadsheet](https://docs.google.com/spreadsheets/d/107jqrutZhU0fVZJ19SBqxxVKbV2NWSdQC9MFYdLiAxc/edit?usp=sharing) to reflect any new or changed analytics.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">View files have been modified, but no screenshot or video is included in the pull request. Consider adding some for clarity.</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: podcasts/ProfileViewController.swift [hunk 1]
```diff
@@ -500,7 +500,7 @@ class ProfileViewController: PCViewController, UITableViewDataSource, UITableVie
 
     @objc private func referralsTapped() {
         hideReferralsHint()
-        let viewModel = ReferralSendPassModel(offerInfo: referralsOfferInfo,
+        let viewModel = ReferralSendPassModel(offerInfo: referralsOfferInfo, referralURL: URL(string: "https://pocketcasts.com"),
```
Comment (danielebogo) [inline podcasts/ProfileViewController.swift]: Nit: we have a Server constant called `pocketcastsDotCom` -> `ServerConstants.Urls.pocketcastsDotCom`. We use it in few spot like the sharing helper.

File: podcasts/Referrals/ReferralSendPassView.swift [hunk 2]
```diff
@@ -66,6 +80,6 @@ struct ReferralSendPassView: View {
 
 #Preview("With Passes") {
     Group {
-        ReferralSendPassView(viewModel: ReferralSendPassModel(offerInfo: ReferralsOfferInfoMock()))
+        ReferralSendPassView(viewModel: ReferralSendPassModel(offerInfo: ReferralsOfferInfoMock(), referralURL: URL(string: "https://pocketcasts.com")))
```
Comment (danielebogo) [inline podcasts/Referrals/ReferralSendPassView.swift]: Same here about the URL

**Review by danielebogo — CHANGES_REQUESTED** [review]
Hey @SergioEstevao works as described.
I just left few comments about the URL which can be replaced by a constant we already have

File: podcasts/ProfileViewController.swift (hunk 1 above)
Comment (SergioEstevao) [reply podcasts/ProfileViewController.swift]: Good catch, but the URL here will be something defined by the server for each user... so this was a temporary use. But I updated it [here](https://github.com/Automattic/pocket-casts-ios/pull/2166/commits/0f55838c8c48374c404731227546613ec96d25bf) to use the constant for now.

File: podcasts/Referrals/ReferralSendPassView.swift (hunk 2 above)
Comment (SergioEstevao) [reply podcasts/Referrals/ReferralSendPassView.swift]: Good catch, but the URL here will be something defined by the server for each user... so this was a temporary use. But I updated it [here](https://github.com/Automattic/pocket-casts-ios/pull/2166/commits/0f55838c8c48374c404731227546613ec96d25bf) to use the constant for now.

**Comment by SergioEstevao** [comment]
@danielebogo can you give it another look

**Review by danielebogo — APPROVED** [review]
:shipit:

**Pull request merged**